### DISCUSSION TITLE
Unable to change governor

### DISCUSSION THREAD

**Discussion by bttd** [tteck/Proxmox#1139] [Help Needed]
Hi,

I try to change the CPU governor with your script on my proxmox installation. It tells me to choose from menu, but how can I do it? 

I'm unable to navigate between the listed items, and if I type the number of the governor or the name, It doesn't change anything. 

Also, it is possible to call the script with predefined governor? I want to set up a cronjob for every reboot. 

Regards

**Comment by tteck** [comment]
```
   __________  __  __
  / ____/ __ \/ / / /
 / /   / /_/ / / / / 
/ /___/ ____/ /_/ /  
\____/_/    \____/   
Scaling Governors

Proxmox IP 192.168.86.3 
Current Kernel 6.1.15-1-pve

Available CPU Scaling Governors
performance powersave

Current CPU Scaling Governor
performance

** 1) performance CPU Scaling Governor
** 2) powersave CPU Scaling Governor

NOTE: Settings return to default after reboot

Please choose an option from the menu and press [ENTER] or x to exit.
```
In this case, you'd type **2** then press **enter** to select the powersave CPU Scaling Governor

**Reply by tteck** [reply]
Watch the "Current CPU Scaling Governor", it will change.

**Reply by bttd** [reply]
Ok, I check it again. I have this list: 

** 1) conservative CPU Scaling Governor
** 2) ondemand CPU Scaling Governor
** 3) userspace CPU Scaling Governor
** 4) powersave CPU Scaling Governor
** 5) performance CPU Scaling Governor
** 6) schedutil CPU Scaling Governor

If I choose 1 or 2 nothing happend it stay performance

Only userspace, powersave and performace works.

**Reply by tteck** [reply]
What's showing in the "Available CPU Scaling Governors"?

**Reply by bttd** [reply]
Current Kernel 5.15.85-1-pve

Available CPU Scaling Governors
conservative ondemand userspace powersave performance schedutil 

Current CPU Scaling Governor
userspace

** 1) conservative CPU Scaling Governor
** 2) ondemand CPU Scaling Governor
** 3) userspace CPU Scaling Governor
** 4) powersave CPU Scaling Governor
** 5) performance CPU Scaling Governor
** 6) schedutil CPU Scaling Governor

NOTE: Settings return to default after reboot

Please choose an option from the menu and press [ENTER] or x to exit.

**Reply by bttd** [reply]
Few more details:

Syslog, if I try to change to ondemand:
`Mar 10 12:46:06 office kernel: [1453322.400440] cpufreq: Can't use ondemand governor as dynamic switching is disallowed. Fallback to performance governor`

The hw is a Dell DL380 G8 with Intel(R) Xeon(R) CPU X5677

**Reply by tteck** [reply]
I don't know that CPU, but it sounds like `intel_pstate` is active.
What's the output from `cat /sys/devices/system/cpu/intel_pstate/status`

**Reply by bttd** [reply]
After I disable power management in server bios, it start working. 
Thanks for the support!

Is there any way, to set a cron job for reboot with this script?

**Reply by tteck** [reply]
If you want to set the CPU scaling governor to "ondemand" upon every reboot, you can add a cron job by running `crontab -e` as root, and adding the following line:
```
@reboot echo "ondemand" | tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor
```

To monitor the CPU frequency in real-time, with updates every two seconds, you can use the `watch` command to run `lscpu` and filter for the "MHz" field:
```
watch "lscpu | grep MHz"
```
